backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/validator/AnsibleRunnerCleanUpService.java

```diff
@@ -0,0 +1,78 @@
+package org.ovirt.engine.core.bll.validator;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.concurrent.ManagedScheduledExecutorService;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.ovirt.engine.core.common.BackendService;
+import org.ovirt.engine.core.common.config.Config;
+import org.ovirt.engine.core.common.config.ConfigValues;
+import org.ovirt.engine.core.common.utils.ansible.AnsibleConstants;
+import org.ovirt.engine.core.utils.threadpool.ThreadPools;
+
+@Singleton
+public class AnsibleRunnerCleanUpService implements BackendService {
+
+    private final String executionDate = "execution_date.txt";
```

Comment:
Please instead of the file using some file created that contains the timestamp use the standard `BasicFileAttributes`
Example:
```
BasicFileAttributes attr = Files.readAttributes(file, BasicFileAttributes.class);
System.out.println("creationTime: " + attr.creationTime());
System.out.println("lastAccessTime: " + attr.lastAccessTime());
System.out.println("lastModifiedTime: " + attr.lastModifiedTime());
```

or if you don't wanna use the `BasicFileAttributes` you can use something like:
```
File file = new File(fileName);
System.out.println("Before Format : " + file.lastModified());
SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
System.out.println("After Format : " + sdf.format(file.lastModified()));
```

Reply:
I'm not sure if relying on filesystem attributes is a good option, those can be changed accidentally. Using our file with timestamp written in it is much more protected against accidental change

Reply:
If we would put users' interaction into perspective, they can also edit/remove the timestamp which can cause failure, or engine issues can cause not creating the timestamp.

Reply:
I still think that the file timestamp is the best solution, if the user would edit some files they would be just cleaned after 2 weeks.
Plus the file is no longer created after the code cleanup. https://github.com/oVirt/ovirt-engine/pull/231/files#diff-c14903676ceca0238e0ca76b26dfbb0e8c8f8786ad5ef17ced6d56f98e01c2fdL310

Reply:
OK, so let's use for now the creation date of `/var/lib/ansible-runner/<UUID>` to determine if the directory should be cleaned up

Reply:
done